thunder/executors/cudnn_sdpa.py

```diff
@@ -0,0 +1,759 @@
+    if globals().get("cudnn", None) is None:
+        return False
+
+    if query.device.type != "cuda" or key.device != query.device or value.device != query.device:
+        return False
+
+    if len(query.size()) != 4:
+        return False
+    _, _, _, d_q = query.size()
+
+    if len(value.size()) != 4:
+        return False
+    _, _, _, d_kv = value.size()
+
+    # Bug in cudnn 8.9.5 and earlier where embedding dim support is missing
+    for d in [d_q, d_kv]:
+        if d % 8 != 0 or d > 128:
+            return False
+
+    dropout_p = pyval(dropout_p)
+    is_causal = pyval(is_causal)
+    if scale is not None:
+        scale = pyval(scale)
+    try:
+        # TensorProxy do not contain stride information, but cudnn graph requires them.
+        # Assume row major layout for now. If the strides during execution are different, a new graph will be built.
+        query_stride = _compute_row_major_strides(query.size())
+        key_stride = _compute_row_major_strides(key.size())
+        value_stride = _compute_row_major_strides(value.size())
+
+        if attn_mask is not None:
+            # Make attn_mask to be of the same dimensionality as other input tensors
+            attn_mask_shape = (1,) * (query.ndim - attn_mask.ndim) + attn_mask.shape
+            # cudnn does not support boolean attn_mask, so make it additive mask instead.
+            # During execution, similar change to attn_mask buffer will be made, where all values of False will be replaced with -inf
+            attn_mask_dtype = query.dtype if attn_mask.dtype in [torch.bool, dtypes.bool8] else attn_mask.dtype
+            attn_mask = TensorProxy(like=attn_mask, shape=attn_mask_shape, dtype=attn_mask_dtype)
+
+        # Build both forward and backward graphs
+        _make_cudnn_sdpa_forward_graph(
+            query, key, value, attn_mask, dropout_p, is_causal, query_stride, key_stride, value_stride
+        )
+        _make_cudnn_sdpa_backward_graph(
+            query,
+            key,
+            value,
+            attn_mask,
+            dropout_p,
+            is_causal,
+            query_stride,
+            key_stride,
+            value_stride,
+            query_stride,
+            key_stride,
+            value_stride,  # Use the same strides as inputs for their respective grads
+        )
+    # If cudnn can't support the graph, return false
+    # Please turn on cudnn API logging for helpful messages that mention why the graph is not supported.
+    # For cudnn backend logging, refer https://docs.nvidia.com/deeplearning/cudnn/latest/reference/troubleshooting.html
+    # For cudnn frontend logging, refer https://github.com/NVIDIA/cudnn-frontend?tab=readme-ov-file#debugging
+    except cudnn.cudnnGraphNotSupportedError:
+        return False
+    # Otherwise just raise the error.
+    # These errors can be due to internal cudnn bugs, or user error.
+    except Exception:
+        raise
+
+    return True
+
+
```

Comment:
Also here the registration for the fwd should be here

```suggestion
cudnn_sdpa_fwd = cudnn_ex.register_operator(
    "cudnn_sdpa_fwd",
    meta=_cudnn_sdpa_forward_meta,
    fn=_cudnn_sdpa_fwd_impl,
    tags=(OpTags.DONT_AUTO_RECOMPUTE_IN_BACKWARD,),
)


```